### PR TITLE
Dispatch post write event processor

### PR DESCRIPTION
**Pull request by loic425**
| Q               | A
| --------------- | -----
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | 
| License         | MIT

Based on https://github.com/Sylius/SyliusResourceBundle/pull/828
Extracted from https://github.com/Sylius/SyliusResourceBundle/pull/817

### CODE REVIEW

File: src/Component/src/Symfony/EventDispatcher/State/DispatchPostWriteEventProcessor.php [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        private OperationEventHandlerInterface $eventHandler,
+    ) {
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function process(mixed $data, Operation $operation, Context $context): mixed
+    {
+        $data = $this->processor->process($data, $operation, $context);
```
Comment (diimpp) [inline src/Component/src/Symfony/EventDispatcher/State/DispatchPostWriteEventProcessor.php:39]: Is it alright for both `PreWrite` and `PostWrite` to call "inner" processor? Won't it result on double execution?

File: src/Component/src/Symfony/EventDispatcher/State/DispatchPostWriteEventProcessor.php [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+        $operationEvent = $this->operationEventDispatcher->dispatchPostEvent($data, $operation, $context);
```
Comment (diimpp) [inline src/Component/src/Symfony/EventDispatcher/State/DispatchPostWriteEventProcessor.php:41]: From naming point of view, does `$this->operationEventDispatcher->dispatchPostEvent` dispatches it for listeners to listen or it's only a event object factory?

Generally speaking, I cannot understand what's going on with "dispatch event" and "handle event" in two consequent lines of code without reading internals of both.

**Review by diimpp — APPROVED** [review]

File: src/Component/src/Symfony/EventDispatcher/State/DispatchPostWriteEventProcessor.php (hunk 1 above)
Comment (loic425) [reply src/Component/src/Symfony/EventDispatcher/State/DispatchPostWriteEventProcessor.php:39]: No, I've checked and that was ok. The harder thing is to create tests for that and I have no idea.

File: src/Component/src/Symfony/EventDispatcher/State/DispatchPostWriteEventProcessor.php (hunk 2 above)
Comment (loic425) [reply src/Component/src/Symfony/EventDispatcher/State/DispatchPostWriteEventProcessor.php:41]: Yeah I do not like this part, and that's only to allow developers to switch from resource controller to new operations system easily.

On resource controller:
Pre write
https://github.com/Sylius/SyliusResourceBundle/blob/1.11/src/Bundle/Controller/ResourceController.php#L175
Post write
https://github.com/Sylius/SyliusResourceBundle/blob/1.11/src/Bundle/Controller/ResourceController.php#L202

Comment (loic425) [reply src/Component/src/Symfony/EventDispatcher/State/DispatchPostWriteEventProcessor.php:41]: In the future, the idea is to handle this with your custom processor or decorating a Sylius one in the e-commerce part when Sylius will use operations.

**Review by oallain — APPROVED** [review]

**Review by GSadee — APPROVED** [review]

**Comment by GSadee** [comment]
Thanks, Loïc! :tada:

**Pull request merged**